src/functions/schema.ts

```diff
@@ -65,7 +81,8 @@ function getSchemaId(schemaObj: JSONSchema): void | string {
 }
 
 const validators = new (class extends WeakMap<JSONSchema, ValidateFunction> {
-  public get(schemaObj: JSONSchema) {
+  public get(schemaObj: JSONSchema, oasVersion?: Optional<number>) {
+    const ajv = getAjv(oasVersion);
```

Comment:
@P0lip how cheap is it to recreate a new instance in that layer?

Reply:
Only three instances are created, one for each "schema version" (i.e. oas2, oas3 and json schema). So overhead should be minimal and better than creating one ajv instance per example validation.

Reply:
Yeah, it should be cheap.